### ISSUE TITLE
Unable to launch session

### ISSUE THREAD

**Issue by AnAkIn1** [eliasp/plasma-workspace-units#19]
I've now tried with systemd 223, kdbus for kernel 4.1 and latest plasma-workspace-units, and I am unable to launch a session at all. SDDM just freeze when I click login, and after a while everything becomes black.

I guess this is related to the following errors:

systemd[1294]: kded.service: Two services allocated for the same bus name org.kde.kded5, refusing operation.
systemd[1294]: kwalletd.service: Two services allocated for the same bus name org.kde.kwalletd5, refusing operation.

I've modified https://aur4.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=plasma-workspace-units-git to remove the commit number, remove the dbus file line and change dbus/session.d/* to dbus/services/*

**Comment by AnAkIn1** [comment]
I had to delete a lot of files provided by plasma packages in /usr/share/dbus-1/services to solve this because they have the same bus name, as the error message says.

**Comment by eliasp** [comment]
@bhush9 Placing the dbus service files in `dbus/session.d/*` was an earlier mistake on my side. They need to go to `/usr/share/dbus-1/services`. There they will have to override the existing files there - I don't know how Arch/pacman handles such conflicts.

We should get `SystemdService=` in the DBus service files upstream ASAP as this won't have any side-effects on non systemd/kdbus users/systems and then we won't have to deal with this here anymore.

**Comment by AnAkIn1** [comment]
pacman can't handle such conflicts, the package won't install unless you use --force to ignore file conflicts.

**Comment by AnAkIn1** [comment]
Note that there also seem to be a conflict between kf5_org.kde.kuiserver.service (plasma-workspace) and org.kde.kuiserver.service (kdebase-runtime) which produce the same error as they have the same bus name. I guess something will need to be done about that too.

**Comment by bhush9** [comment]
Yes, ideally org.kde.kuiserver.service file from here will be removed, and SystemdUnit= line will be added in kf5_org.kde.kuiserver.service upstream

**Comment by AnAkIn1** [comment]
I think the problem is that upstream provide these two files with the same bus name.

**Comment by bhush9** [comment]
Well, no I guess kdebase-runtime conflicts with plasma-workspace so it is not problem.

**Comment by AnAkIn1** [comment]
Not on arch, is it needed for some KDE4 applications to work with Plasma 5. Maybe the packager can remove the file though.

**Comment by bhush9** [comment]
@AnAkIn1 This looks like problem that should be reported to KDE, Seems they just managed file name conflict but not dbus service name conflict.. :laughing:

**Comment by bhush9** [comment]
I will take care of it

**Comment by bhush9** [comment]
https://git.reviewboard.kde.org/r/117876/ : Same concerns were raised when renaming..

**Comment by AnAkIn1** [comment]
Any chance these files could make it into the next Plasma/Frameworks releases?

**Comment by eliasp** [comment]
Unlikely for 5.4 as the freeze is very close, but 5.5 should be no problem. We're continuing now the work in an upstream repo from where we want to slowly move all required pieces into their target components.

https://projects.kde.org/projects/playground/base/plasma-systemd-integration
